syntax = "proto3";

import "shared.proto";
import "entity.proto";

package quran;

message RevelationRequest {
    string revelation_type = 1;
}

message SurahSingleResponse {
    int32 code = 1;
    string status = 2;
    SurahEntity data = 3;
    string message = 4;
}

message SurahMultiResponse {
    int32 code = 1;
    string status = 2;
    repeated SurahEntity data = 3;
}

service Surah {
    rpc CreateSurah (SurahEntity) returns (SurahSingleResponse) {}
    rpc GetAll (EmptyMessage) returns (SurahMultiResponse) {}
    rpc FindSurahById (IDRequest) returns (SurahSingleResponse) {}
    rpc FindSurahByNumber (NumberRequest) returns (SurahSingleResponse) {}
    rpc FindSurahByName (NameRequest) returns (SurahSingleResponse) {}
    rpc FindSurahByEnglishName (NameRequest) returns (SurahSingleResponse) {}
    rpc FindSurahByRevelationType (RevelationRequest) returns (SurahMultiResponse) {}
}